internal/coreinternal/scraperinttest/scraperint.go

```diff
@@ -167,57 +189,70 @@ func WithExpectedFile(f string) TestOption {
 	}
 }
 
-func WithCompareOptions(opts ...pmetrictest.CompareMetricsOption) TestOption {
+func WriteExpected() TestOption {
 	return func(it *IntegrationTest) {
-		it.compareOptions = opts
+		it.writeExpected = true
 	}
 }
 
-func WithCompareTimeout(t time.Duration) TestOption {
+func WithCompareOptions(opts ...pmetrictest.CompareMetricsOption) TestOption {
 	return func(it *IntegrationTest) {
-		it.compareTimeout = t
+		it.compareOptions = opts
 	}
 }
 
-func WriteExpected() TestOption {
+func WithCompareTimeout(t time.Duration) TestOption {
 	return func(it *IntegrationTest) {
-		it.writeExpected = true
+		it.compareTimeout = t
 	}
 }
 
 type customConfigFunc func(*testing.T, component.Config, *ContainerInfo)
 
 type ContainerInfo struct {
-	host  string
-	ports map[string]string
+	sync.Mutex
+	containers map[string]testcontainers.Container
 }
 
-func (c ContainerInfo) Host(t *testing.T) string {
-	require.NotEmpty(t, c.host, "container not in use")
-	return c.host
+func (ci *ContainerInfo) Host(t *testing.T) string {
+	return ci.HostForNamedContainer(t, "")
```

Comment:
Is `""` a valid container name or will this always cause the downstream `require.True(t, ok, "container with name %q not found", name)` to fail?

Reply:
`""` is the default name, so it works unless you explicitly specify a name.